### PR TITLE
Provide config service methods to java clients

### LINKED ISSUE #3528: Java Client doesn't have a ConfigService as it does other services
At this time, it appears that the ConfigService grpc is only for internal use, the API isnt exposed at all, except as a plain grpc call.

Instead, consider a ConfigService in io.deephaven.client.impl:
```java
public interface ConfigService {
  CompletableFuture<Map<String, ConfigValue>> getAuthenticationConstants();
  CompletableFuture<Map<String, ConfigValue>> getConfigurationConstants();
}
```

### PR DESCRIPTION
**Pull request by niloc132**
Refactors existing java-client grpc response observers into one shared implementation, so that all can be canceled.

Fixes #3528

### CODE REVIEW

File: java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+/**
+ * ClientResponseObserver implementation that can map results into a CompletableFuture. To keep generics under control,
+ * has two factory methods, {@link #of(Object, BiConsumer, Function)} and {@link #ignoreResponse(Object, BiConsumer)}.
+ * 
+ * @param <ReqT> The type of the gRPC request
+ * @param <RespT> The type of the gRPC response
+ * @param <V> The type of the result to make available in the future
+ */
+public class UnaryGrpcFuture<ReqT, RespT, V> implements ClientResponseObserver<ReqT, RespT> {
```
Comment (devinrsmith) [inline java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java:19]: Is there value in making this public? (I know we are very loose in other places w/ public...)

File: java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+     * @param <V> The type of the result to make available in the future
+     * @return A CompletableFuture that will succeed, fail, or be canceled. If it succeeds, will contain a result from
+     *         the mapping function.
+     */
+    public static <ReqT, RespT, V> CompletableFuture<V> of(ReqT request, BiConsumer<ReqT, StreamObserver<RespT>> call,
+            Function<RespT, V> mapping) {
+        UnaryGrpcFuture<ReqT, RespT, V> observer = new UnaryGrpcFuture<>(mapping);
+        call.accept(request, observer);
+        return observer.future();
+    }
```
Comment (devinrsmith) [inline java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java:38]: What are your thoughts on having the callers always get `CompletableFuture<RespT>` back here (ie, no `mapping` arg), and they can call `.thenApply(...)` as appropriate?

**Review by devinrsmith — CHANGES_REQUESTED** [review]
Generally nice abstraction. One big issue, `subscribeToFields` broken.

File: java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java (hunk 2 above)
Comment (niloc132) [reply java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java:38]: I actually started off that way, as it seems nicer to be "fluent" like that.  The problem there is that if they return the result of `.thenApply()` or whatnot, then the caller is left with the downstream CF, and canceling that won't cancel the upstream one (so won't stop the grpc call).

File: java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java (hunk 1 above)
Comment (niloc132) [reply java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java:19]: My assumption was that just as DeephavenChannel is very-public (presumably to support the various use cases that java-client doesnt have implemented?), we want this tool available for downstream users to make unary calls easy to build with CFs.

File: java-client/session/src/main/java/io/deephaven/client/impl/SessionImpl.java [hunk 3]
```diff
@@ -257,17 +261,15 @@ public CompletableFuture<Void> deleteFromInputTable(HasTicketId destination, Has
...
 
     @Override
     public Cancel subscribeToFields(Listener listener) {
-        final ListFieldsRequest request = ListFieldsRequest.newBuilder().build();
-        final ListFieldsObserver observer = new ListFieldsObserver(listener);
-        bearerChannel.application().listFields(request, observer);
-        return observer;
+        CompletableFuture<Void> future = UnaryGrpcFuture
+                .ignoreResponse(ListFieldsRequest.getDefaultInstance(), channel().application()::listFields);
+        return () -> future.cancel(false);
```
[the remark this answers is not in the document]
Comment (niloc132) [reply java-client/session/src/main/java/io/deephaven/client/impl/SessionImpl.java:272]: Agreed, I'll undo this.

File: java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java (hunk 2 above)
Comment (devinrsmith) [reply java-client/session/src/main/java/io/deephaven/client/impl/UnaryGrpcFuture.java:38]: Good call. While the caller _could_ still add their own `.whenComplete(...)` and pass along the cancel, that's a bunch of extra machinery I agree we don't really want.

**Review by devinrsmith — APPROVED** [review]

**Review by abaranec — APPROVED** [review]

**Pull request merged**